### PR TITLE
Fix use of deprecated configs 'spring.data.neo4j.*'

### PR DESCRIPTION
**Pull request by abelsromero**
* Replace deprecated 'spring.data.neo4j.*' by 'spring.neo4j.*'
* Add how-to run neo4j server in Docker for all OSs
* Several README fixes

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -36,15 +36,15 @@ This service pulls in all the dependencies you need for an application and does
...
 
-NOTE: You can also fork the project from Github and open it in your IDE or other editor.
+NOTE: You can also fork the project from GitHub and open it in your IDE or other editor.
 
 == Standing up a Neo4j Server
 
 Before you can build this application, you need to set up a Neo4j server.
 
-Neo4j has an open source server you can install for free.
+Neo4j has an open source server you can install for free, or run it with Docker.
```
Comment (Buzzardo) [inline README.adoc:45]: Add "you can" between "or" and "run".

File: README.adoc [hunk 2]
```diff
@@ -15,7 +15,7 @@ build an embedded Neo4j server, store entities and relationships, and develop qu
 
 == What You Need
 
-:java_version: 1.8
+:java_version: 17
```
Comment (Buzzardo) [inline README.adoc:18]: Thanks for catching that.

**Review by Buzzardo — CHANGES_REQUESTED** [review]

File: README.adoc [hunk 3]
```diff
@@ -88,6 +88,21 @@ curl -v -u neo4j:neo4j POST localhost:7474/user/neo4j/password -H "Content-type:
 This changes the password from `neo4j` to `secret` -- something to NOT do in production!
 With that step completed, you should be ready to run the rest of this guide.
 
+Alternatively, to run with the https://hub.docker.com/_/neo4j[neo4j Docker image].
```
[the remark this answers is not in the document]
Comment (abelsromero) [reply README.adoc]: Done, thanks a lot!

File: README.adoc (hunk 1 above)
Comment (abelsromero) [reply README.adoc]: Done, thanks a lot!

**Review by Buzzardo — APPROVED** [review]

**Comment by Buzzardo** [comment]
Thanks, Abel. :)

**Pull request merged**